pkg/i2gw/provider.go

```diff
@@ -99,8 +101,94 @@ type GatewayResources struct {
 	UDPRoutes  map[types.NamespacedName]gatewayv1alpha2.UDPRoute
 
 	ReferenceGrants map[types.NamespacedName]gatewayv1beta1.ReferenceGrant
+
+	GatewayExtensions map[types.NamespacedName]unstructured.Unstructured
+}
+
+// IR contains all Gateway-API objects, and Intermediate Representation(IR) to
+// hold specicifications on Gateway Objects for supporting Ingress extension
+// features. An feature on IR can be mapped to changes on Gateway-API objects,
+// or provider-specific Gateway extensions.
```

Comment:
```suggestion
// IR holds specifications of Gateway Objects for supporting Ingress extensions, annotations, and proprietary api
// features not supported as Gateway core features. An IR field can be mapped to core Gateway-API fields,
// or provider-specific Gateway extensions.
```

Reply:
Sure, just created an example for session affinity feature translation, PTAL!